framework/Models/PostProcessors/DataClassifier.py

```diff
@@ -93,133 +93,93 @@ def _handleInput(self, paramInput):
-              if newInput[dataType]['input'][elem] is None:
-                newInput[dataType]['input'][elem] = np.empty(0)
-              newInput[dataType]['input'][elem] = np.append(newInput[dataType]['input'][elem], rlz[elem])
-            for elem in outputParams:
-              if newInput[dataType]['output'][elem] is None:
-                newInput[dataType]['output'][elem] = np.empty(0)
-              newInput[dataType]['output'][elem] = np.append(newInput[dataType]['output'][elem], rlz[elem].values[-1])
-              if newInput[dataType]['historySizes'][i] is None:
-                newInput[dataType]['historySizes'][i] = len(rlz[elem].values)
-
+          self.raiseAnError(IOError, "None of the input DataObjects can be used as the reference classifier! Either the label", \
+                  self.label, "is not exist in the output of the DataObjects or the inputs of the DataObjects are not the same as", \
+                  ','.join(self.mapping.keys()))
+      newInput[dataType] = inputDict
     return newInput
 
   def run(self, inputIn):
     """
       This method executes the postprocessor action.
-      @ In,  inputIn, list, list of DataObjects
+      @ In,  inputIn, list, list of input dictionaries
       @ Out, outputDict, dict, dictionary of outputs
     """
-    inputDict = self.inputToInternal(inputIn)
+    inputDict = self.identifyInputs(inputIn)
     targetDict = inputDict['target']
     classifierDict = inputDict['classifier']
     outputDict = {}
     outputDict.update(inputDict['target']['data'])
     outputType = targetDict['type']
-    numRlz = utils.first(targetDict['input'].values()).size
+    dimsDict = targetDict['dims']
+    numRlz = utils.first(targetDict['data'].values()).size
     outputDict[self.label] = []
     for i in range(numRlz):
       tempTargDict = {}
-      for param, vals in targetDict['input'].items():
-        tempTargDict[param] = vals[i]
-      for param, vals in targetDict['output'].items():
+      for param, vals in targetDict['data'].items():
         tempTargDict[param] = vals[i]
       tempClfList = []
       labelIndex = None
-      for key, values in classifierDict['input'].items():
+      for key in self.mapping.keys():
         calcVal = self.funcDict[key].evaluate("evaluate", tempTargDict)
+        values = classifierDict['data'][key]
         inds, = np.where(np.asarray(values) == calcVal)
         if labelIndex is None:
           labelIndex = set(inds)
         else:
           labelIndex = labelIndex & set(inds)
       if len(labelIndex) != 1:
-        self.raiseAnError(IOError, "The parameters", ",".join(tempTargDict.keys()), "with values", ",".join([str(el) for el in tempTargDict.values()]), "could not be put in any class!")
-      label = classifierDict['output'][self.label][list(labelIndex)[0]]
+        self.raiseAnError(IOError, "The parameters", ",".join(tempTargDict.keys()), "with values", ",".join([str(el) for el in tempTargDict.values()]), "could not be classified!")
```

Comment:
Hm, I suppose we could get this to pass the precheck by changing to "could not be classif"+"ied!"

Reply:
Maybe the term `categorized` could be swapped as a synonym?

Reply:
changed to "categorized", and the precheck is good with that.